File: tarefas/labs/Concurso.java
import java.util.Scanner;

public class Concurso{
    public static class Atleta{
        private int peso;
        private String nome;

        Atleta(String nome, int peso){
            this.nome = nome;
            this.peso = peso;
        }

        public int getPeso(){
            return this.peso;
        }
        public String getNome(){
            return this.nome;
        }

        public void imprimir(){
            System.out.println(this.nome + " " + this.peso);
        }
    }
    public static void ordenarLista(Atleta[] lista){
        int n = lista.length;

        for(int i = 0; i < n - 1; i++){
            for(int j = 0; j < n - 1; j++){
                Atleta atletaEsq = lista[j];
                Atleta atletaDir = lista[j + 1];
               if(atletaEsq.getPeso() < atletaDir.getPeso()){
                    Atleta temp = lista[j];
                    lista[j] = lista[j + 1];
                    lista[j + 1] = temp;
               }else if(atletaEsq.getPeso() == atletaDir.getPeso() && atletaEsq.getNome().compareTo(atletaDir.getNome()) > 0){
                    Atleta temp = lista[j];
                    lista[j] = lista[j + 1];
                    lista[j + 1] = temp;
               }
            }
        }
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        Atleta[] listaAtletas = new Atleta[n];

        for(int i = 0; i < n; i++){
            String nomeAtleta = sc.next();
            int pesoAtleta = sc.nextInt();

            listaAtletas[i] = new Atleta(nomeAtleta, pesoAtleta);
        }
        ordenarLista(listaAtletas);
        
        for(int i = 0; i < n; i++){
            listaAtletas[i].imprimir();
        }

        sc.close();
    }

}

File: tarefas/labs/Concurso.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct Atleta
{
    int peso;
    char nome[51];
} Atleta;

int main()
{

    int n = 0;

    scanf("%d", &n);

    Atleta listaAtletas[n];

    for (int i = 0; i < n; i++)
    {
        scanf("%s %d", listaAtletas[i].nome, &listaAtletas[i].peso);
    }

    for (int i = 0; i < n - 1; i++)
    {
        for (int j = 0; j < n - 1; j++)
        {
            Atleta atletaEsq = listaAtletas[j];
            Atleta atletaDir = listaAtletas[j + 1];
            if (atletaEsq.peso < atletaDir.peso)
            {
                Atleta temp = listaAtletas[j];
                listaAtletas[j] = listaAtletas[j + 1];
                listaAtletas[j + 1] = temp;
            }
            else if (atletaEsq.peso == atletaDir.peso && strcmp(atletaEsq.nome, atletaDir.nome) > 0)
            {
                Atleta temp = listaAtletas[j];
                listaAtletas[j] = listaAtletas[j + 1];
                listaAtletas[j + 1] = temp;
            }
        }
    } // end bubbleSort

    for (int i = 0; i < n; i++)
    {
        printf("%s %d\n", listaAtletas[i].nome, listaAtletas[i].peso);
    }

    return 0;
}
